spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phrase, such as dev, testing, live, etc. 
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice, e.g.
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile%`, (e.g. `%live%vechile.name`).
+
+For an instance, a config source can be specified as follows.
+
+[source, text]
+----
+%dev%vehicle.name=car
+%live%vehicle.name=train
+%testing%vehicle.name=bike
+----
+
+In the above config source, if the system property `mp.config.profile` is set to `dev`. The `vechileName` in the following snippet will be resolved to be `car`.
```

Comment:
I think we need to clarify what happens if you activate a profile and search for a key that doesn't have a profile value. Expected behaviour will be to fallback to the key without profile.

Reply:
This is resolved now that we've added this section:

> Conforming implementations are required to search for a configuration source with the highest ordinal (priority) that provides either the property name or the "profile-specific" property name. 
> If the configuration source provides the "profile-specific" name, the value of the "profile-specific" property will be returned. If it doesn't contain the "profile-specific" name, the value of the plain property will be returned. 
> A "profile-specific" property name is defined as a property whose name consists of the following sequence: `% <profile name>.<original property name>`.